bindata/etcd/pod.yaml

```diff
@@ -25,13 +25,15 @@ spec:
           : "${NODE_NODE_ENVVAR_NAME_ETCD_NAME?not set}"
           : "${NODE_NODE_ENVVAR_NAME_IP?not set}"
 
-          if [ "${NODE_NODE_ENVVAR_NAME_IP}" != "${NODE_IP}" ]; then
+          # check for ipv4 addresses as well as ipv6 addresses with extra square brackets
+          if [ "${NODE_NODE_ENVVAR_NAME_IP}" != "${NODE_IP}" -a "${NODE_NODE_ENVVAR_NAME_IP}" != "[${NODE_IP}]" ]; then
```

Comment:
small nit use of `-a` syntax has been dedicated please use `&&`[1].

I think we should add shellcheck to our unit tests.

[1] : https://github.com/koalaman/shellcheck/wiki/SC2166

Reply:
Why is `$NODE_IP` bracket wrapped?

I think it was a reasonable assumption in the original PR that a variable named `$NODE_IP` was *actually* just an IP, so I'd be concerned about this problem surfacing again elsewhere.  IMHO, the IP shouldn't be wrapped until it's included in the host portion of a URI.

net.JoinHostPort in go will handle it, and for scripts we have a helper in our env that might be able to be reused:

https://github.com/openshift-metal3/dev-scripts/blob/49a5e1444b24a0f9f1ab7c033f510c3a13255702/utils.sh#L71-L85

Reply:
@stbenjam Here NODE_IP doesn't come bracket wrapped, but NODE_NODE_ENVVAR_NAME_IP is bracket wrapped as it is used the host portion of a URI. We are trying to match these variables to check the validity of the environment, and hence the check in this init container.

Reply:
Oh right, I misread what was being checked! Sorry.

I think the general point is still the same, it's a reasonable assumption for someone looking at a variable with the name "IP" to assume it's an IP, and not a host for a URI.